x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/snapshotlifecycle/package-info.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/**
+ * This is the Snapshot Lifecycle Management (SLM) core package. This package contains the core classes for SLM, including all of the
+ * custom cluster state metadata objects, execution history storage facilities, and the action definitions. For the main SLM service
+ * implementation classes, please see the <tt>ilm</tt> module's <tt>org.elasticsearch.xpack.snapshotlifecycle</tt> package.
+ *
+ * <p>Contained within this specific package are the custom metadata objects and models used through out the SLM service. The names can
+ * be confusing, so it's important to know the differences between each metadata object.
+ *
+ * <p>The {@link org.elasticsearch.xpack.core.snapshotlifecycle.SnapshotLifecyclePolicy} object is the user provided definition of the
+ * SLM policy. This is what a user provides when creating a snapshot policy, and acts as the blueprint for the create snapshot request
+ * that the service launches. It additionally surfaces the next point in time a policy should be executed.
+ *
+ * <p>Lateral to the policy, the {@link org.elasticsearch.xpack.core.snapshotlifecycle.SnapshotInvocationRecord} represents an execution
+ * of a policy. It includes within it the policy name and details about its execution, success or failure.
```

Comment:
Perhaps link to the history package docs here since the record is what is stored in history?

Reply:
I thought that `SnapshotHistoryItem` was what was stored in history

Reply:
You're right, I misread it, nevermind on this then :)